plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -157,29 +158,28 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
 
         Map<String, MLModelStats> modelStats = new HashMap<>();
         if (mlStatsInput.includeModelStats()) {
-            CountDownLatch latch = new CountDownLatch(1);
             boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
-            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
-                for (String modelId : mlStats.getAllModels()) {
-                    if (isSuperAdmin || !hiddenModels.contains(modelId)) {
-                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
-                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
-                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
-                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                                    actionStatsMap.put(entry.getKey(), entry.getValue());
-                                }
+            Set<String> hiddenModels = mlStatsNodesRequest.getHiddenModelIds() != null
+                ? mlStatsNodesRequest.getHiddenModelIds()
+                : Collections.emptySet();
+            log.info("List of hidden models from Stats transport action: {}", Arrays.toString(hiddenModels.toArray()));
```

Comment:
Change to debug level? Stats API may run regularly, looks not necessary to print this

Reply:
I'm removing this in the next commit. This I used to debug, but forgot to remove.